crates/libtiny_tui/src/msg_area/mod.rs

```diff
@@ -67,9 +67,14 @@ impl MsgArea {
     }
 
     pub(crate) fn resize(&mut self, width: i32, height: i32) {
-        self.width = width;
+        self.width = if self.scroll != 0 { width - 1 } else { width };
+        let old_height = self.height;
         self.height = height;
+        let old_total_lines = self.lines_height;
         self.lines_height = None;
+
+        self.total_visible_lines();
```

Comment:
This method name looks like it's going to return something but from this use it seems like it's doing an update? Maybe the name should reflect that, e.g. `update_total_visible_lines` (no idea what the method is doing yet, just describing the problem).